appcues/src/main/java/com/appcues/data/model/StepContainer.kt

```diff
@@ -12,4 +12,9 @@ internal data class StepContainer(
     val presentingTrait: PresentingTrait,
     val contentHolderTrait: ContentHolderTrait,
     val contentWrappingTrait: ContentWrappingTrait,
-)
+) {
+
+    init {
+        steps.forEach { it.parent = this }
```

Comment:
setting "parent" reference on each step makes it easier to reference its container when getting presenting trait.

another idea is to actually pass in the presentingTrait to all children, but setting the parent reference might be useful for other things in the future if we need.

Reply:
scratch that, its so much easier to have presenting trait in the step, also leaving an lateinit var variable not tested from within the class is a bad idea.